File: tests/python/extensions/pybind/core/statevector.py
#!/usr/bin/env python3

import numpy as np
import pybind_isce3 as isce3

def test_sv():
    t = isce3.core.DateTime(2020, 8, 4)
    pos = [0, 0, 1.]
    vel = [1, 0, 0.]
    sv = isce3.core.StateVector(t, pos, vel)
    assert sv.datetime == t
    assert np.allclose(sv.position, pos)
    assert np.allclose(sv.velocity, vel)

File: python/extensions/pybind_isce3/core/StateVector.cpp
#include "StateVector.h"

#include <pybind11/eigen.h>
#include <pybind11/numpy.h>
#include <isce3/core/DateTime.h>
#include <isce3/core/DenseMatrix.h>

namespace py = pybind11;

using namespace isce3::core;

void addbinding(py::class_<StateVector> &pyStateVector)
{
    pyStateVector
        .def(py::init<const DateTime&, const Vec3&, const Vec3&>(),
            py::arg("datetime"), py::arg("position"), py::arg("velocity"))
        .def_readonly("datetime", &StateVector::datetime)
        .def_readonly("position", &StateVector::position)
        .def_readonly("velocity", &StateVector::velocity)
    ;
}
